src/content/en/fundamentals/media/mobile-web-video-playback.md

```diff
@@ -0,0 +1,611 @@
+  &lt;div id="videoControls"&gt;...&lt;/div&gt;
+&lt;/div&gt;
+</pre>
+
+Caution: Set `playsinline` only if you provide your own media controls or show
+native controls with `<video controls>`.
+
+### Toggle fullscreen on button click
+
+Now that we prevent automatic fullscreen, we need to handle ourselves the
+fullscreen mode for the video with the [Fullscreen API]{: .external}. When user
+clicks the "fullscreen button", let's exit fullscreen mode with
+`document.exitFullscreen()` if fullscreen mode is currently in use by the
+document. Otherwise, request fullscreen on the video container with the method
+`requestFullscreen()` if available or fallback to `webkitEnterFullscreen()` on
+the video element only on iOS.
+
+Note: I'm going to use a [tiny shim] for the Fullscreen API in code snippets below that
+will take care of prefixes as the API is not unprefixed yet at that time.
+
+<pre class="prettyprint lang-html">
+&lt;div id="videoContainer">
+  &lt;video id="video" src="file.mp4">&lt;/video&gt;
+  &lt;div id="videoControls">
+    &lt;button id="playPauseButton">&lt;/button>
+    &lt;button id="seekForwardButton">&lt;/button>
+    &lt;button id="seekBackwardButton">&lt;/button>
+    <strong>&lt;button id="fullscreenButton">&lt;/button></strong>
+    &lt;div id="videoCurrentTime">&lt;/div>
+    &lt;div id="videoDuration">&lt;/div>
+    &lt;div id="videoProgressBar">&lt;/div>
+  &lt;/div>
+&lt;/div>
+</pre>
+
+    fullscreenButton.addEventListener('click', function(event) {
+      event.stopPropagation();
+      if (document.fullscreenElement) {
+        document.exitFullscreen();
+      } else {
+        requestFullscreenVideo();
+      }
+    });
+
+    function requestFullscreenVideo() {
+      if (videoContainer.requestFullscreen) {
+        videoContainer.requestFullscreen();
+      } else {
+        video.webkitEnterFullscreen();
+      }
+    }
+
+    document.addEventListener('fullscreenchange', function() {
+      fullscreenButton.classList.toggle('active', document.fullscreenElement);
+    });
+
+<video controls controlsList="nodownload" muted playsinline>
+  <source src="/web/fundamentals/media/videos/toggle-fullscreen-on-button-click.webm"
+          type="video/webm">
+  <source src="/web/fundamentals/media/videos/toggle-fullscreen-on-button-click.mp4"
+          type="video/mp4">
+</video>
+
+### Toggle fullscreen on screen orientation change
+
+As user rotates device in landscape mode, let's be smart about this and
+automatically request fullscreen to create an immersive experience. For this,
+we'll need the [Screen Orientation API]{: .external} which is not yet supported
+everywhere.  Thus, this will be our first progressive enhancement.
+
+How doest this work? As soon as we detect the screen orientation changes, let's
+request fullscreen if the browser window is in landscape mode (that is, its
+width is greater than its height). If not, let's exit fullscreen. That's all.
+
+    if ('orientation' in screen) {
```

Comment:
Either try to support all the implementations but that can be hard. Firefox is easy to test but IE/Edge might be tough. Otherwise, at least mention that there are prefixed APIs if the reader care.